packages/riverpod_lint/lib/src/lints/avoid_use_ref_inside_dispose.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:riverpod_analyzer_utils/riverpod_analyzer_utils.dart';
+
+import '../riverpod_custom_lint.dart';
+
+const disposeMethod = 'dispose';
+
+class AvoidUseRefInsideDispose extends RiverpodLintRule {
+  const AvoidUseRefInsideDispose() : super(code: _code);
+
+  static const _code = LintCode(
+    name: 'avoid_ref_inside_state_dispose',
+    problemMessage: "Avoid using 'Ref' in the dispose method.",
+  );
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addMethodInvocation((node) {
+      final targetType = node.realTarget?.staticType;
+
+      if (targetType == null) return;
+
+      if (widgetRefType.isAssignableFromType(targetType)) {
```

Comment:
Could you use early returns instead of nested conditions?

```suggestion
      if (!widgetRefType.isAssignableFromType(targetType)) return;
```

Reply:
Could you also use the same principle with those thisOrAncestorMatching to remove all the nesting?

Reply:
I did some changes following the early return principle.